app/flatpak-builtins-build-bundle.c

```diff
@@ -127,9 +127,14 @@ get_bundle_appstream_data (GFile        *root,
 
   *result = NULL;
 
-  xmls_dir = g_file_resolve_relative_path (root, "files/share/app-info/xmls");
-  appstream_basename = g_strconcat (name, ".xml.gz", NULL);
-  appstream_file = g_file_get_child (xmls_dir, appstream_basename);
+  appstream_file = g_file_resolve_relative_path (root, "files/share/swcatalog/xml/flatpak.xml.gz");
```

Comment:
As I mentioned in the other PR. This same logic is repeated like 3 times. I think this could be pulled out into a function to just get the correct path in `flatpak-utils.c` perhaps.

Reply:
This has been done